### DISCUSSION TITLE
remove from useFieldArray causing issue for fields

### DISCUSSION THREAD

**Discussion by T41WO** [react-hook-form/react-hook-form#5480] [Q&A] [answered]
Hi,

I have a list of object that renders a dropdown and a textbox as items in that list.

![image](https://user-images.githubusercontent.com/7270970/120711332-1905e800-c474-11eb-9a7f-ab09d3934576.png)

`remove` at a particular index (2 below) causes the items in the list after the index to loose **some** values. 
![image](https://user-images.githubusercontent.com/7270970/120711730-93366c80-c474-11eb-8aed-4613bbc7d57c.png)

Please help, I have looked up documentation for [useFieldArray](https://react-hook-form.com/v6/api#useFieldArray).

P.S. This would have worked before but not now, did something change recently? Using v6.15.2.

Thanks

**Comment by bluebill1049** [comment]
can you provide a codesandbox?

**Reply by T41WO** [reply]
This is close to what I have set in my code. Can't repro the exact issue I am facing but this one is acting weird too.
Removing at index 1 seems to remove index 0.
What am I doing wrong here? 

https://codesandbox.io/s/busy-frost-u9hjg?file=/src/App.js

**Comment by thanh-nguyen-95** [comment, marked as answer]
> https://codesandbox.io/s/busy-frost-u9hjg?file=/src/App.js

```jsx
onClick={(index) => remove(index)}
```

You're actually passing `event` to `remove` instead of the actual `index` of each row.

```jsx
onClick={() => remove(index)}
```

This should fix the problem.

**Reply by bluebill1049** [reply]
thanks @thanh-nguyen-95

**Reply by T41WO** [reply]
Good spot, thank you. This solved the issue above. 

This is closer to what I have though. What are the problems with this implementation.

Thanks again.

https://codesandbox.io/s/long-rain-boz3t?file=/src/App.tsx

**Reply by thanh-nguyen-95** [reply]
You have to pass a key to each item for React to know which one you're updating:

```jsx
controllers: fields.map((f, index) => {
  return {
    key: f.id,
    id: f.id,
    control,
    defaultValue: f.selectedComboBoxKey,
    inputName: `selectedSocialMediaList[${index}].displayText`,
    onItemClick: (): void => {
      return;
    },
    options
  };
}),
```

**Reply by T41WO** [reply]
Thanks! This solved my original issue too. Passing the id as key for each item down the chain!

**Reply by bluebill1049** [reply]
Thank you @thanh-nguyen-95
